text/overview.tex

```diff
@@ -166,20 +166,24 @@ \subsection{The Virtual Machine and Gas}\label{sec:virtualmachineandgas}
 \begin{equation}\label{eq:gasregentry}
   \Z_G \equiv \mathbb{Z}_{-2^{63}\dots2^{63}}\ ,\quad
   \N_G \equiv \mathbb{N}_{2^{64}}\ ,\quad
-  \N_R \equiv \N_{2^{32}}
+  \N_R \equiv \N_{2^{64}}
 \end{equation}
 
 It is left as a rather important implementation detail to ensure that the amount of time taken while computing the function $\Psi(\dots, \gascounter, \dots)$ has a maximum computation time approximately proportional to the value of $\gascounter$ regardless of other operands.
 
-The \textsc{pvm} is a very simple \textsc{risc} \emph{register machine} and as such has 13 registers, each of which is a 32-bit quantity, denoted as $\N_R$, a natural less than $2^{32}$.\footnote{This is three fewer than \textsc{risc-v}'s 16, however the amount that program code output by compilers uses is 13 since two are reserved for operating system use and the third is fixed as zero} Within the context of the \textsc{pvm}, $\registers \in \seq{\N_R}_{13}$ is typically used to denote the registers.
-\begin{align}
-  \mathbb{M} &\equiv \ltuple\isa{\mathbf{V}}{\Y_{2^{32}}}\ts\isa{\mathbf{A}}{\lseq\{\text{W}, \text{R}, \none\}\rseq_{2^{32}}}\rtuple
+The \textsc{pvm} is a very simple \textsc{risc} \emph{register machine} and as such has 13 registers, each of which is a 64-bit quantity, denoted as $\N_R$, a natural less than $2^{64}$.\footnote{This is three fewer than \textsc{risc-v}'s 16, however the amount that program code output by compilers uses is 13 since two are reserved for operating system use and the third is fixed as zero} Within the context of the \textsc{pvm}, $\registers \in \seq{\N_R}_{13}$ is typically used to denote the registers.
+\begin{align}\label{eq:pvmmemory}
+  \mathbb{M} &\equiv \tuple{
+    \isa{\mathbf{V}}{\Y_{2^{32}}},
+    \isa{\mathbf{A}}{\seq{\{\text{W}, \text{R}, \none\}}_p}
+  }\,,\ p = \frac{2^{32}}{\mathsf{Z}_P}\\
+  \mathsf{Z}_P &= 2^{12}
 \end{align}
 
-The \textsc{pvm} assumes a simple pageable \textsc{ram} of 32-bit addressable octets where each octet may be either immutable, mutable or inaccessible. The \textsc{ram} definition $\mathbb{M}$ includes two components: a value $\mathbf{V}$ and access $\mathbf{A}$. If the component is unspecified while being subscripted then the value component may be assumed. Within the context of the virtual machine, $\memory \in \mathbb{M}$ is typically used to denote \textsc{ram}.
+The \textsc{pvm} assumes a simple pageable \textsc{ram} of 32-bit addressable octets situated in pages of $2^p = 4096$ octets where each page may be either immutable, mutable or inaccessible. The \textsc{ram} definition $\mathbb{M}$ includes two components: a value $\mathbf{V}$ and access $\mathbf{A}$. If the component is unspecified while being subscripted then the value component may be assumed. Within the context of the virtual machine, $\memory \in \mathbb{M}$ is typically used to denote \textsc{ram}.
```

Comment:
Just below this text we have (I can't put my comment directly there as GH disallows that for parts of the file that were not changed):

> A page fault (attempt to access some address in \textsc{ram} which is not accessible), \\$\fault\\$. This includes the address at fault.

It'd probably be better/simpler to give out the address of the *page* that is at fault (or the page index, if you prefer), and not the exact address that was accessed (considering the program will have to page in the whole page anyway)? Implementation-wise we *can* figure out the exact address that was accessed, but that could have some portability issues on different/future OSes or hardware architectures.